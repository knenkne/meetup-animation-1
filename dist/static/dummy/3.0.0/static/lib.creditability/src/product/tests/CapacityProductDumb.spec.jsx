import React from 'react'
import { shallow } from 'enzyme'
import { Link } from '@sbol/lib.ui'

import { CapacityProductDumb, getPathImage } from '../CapacityProductDumb'
import { STATUS } from '../status'

const version = process.env.VERSION

describe('<CapacityProduct />', () => {
    describe('<CapacityProductDumb />', () => {
        it('should proper select background image', () => {
            expect(getPathImage(STATUS.LC_ACCEPTED)).toEqual('/common//img/statuses/female-done')
            expect(getPathImage(STATUS.LC_CALCULATION_IN_PROGRESS)).toEqual('/common//img/statuses/female-waiting')
            expect(getPathImage(STATUS.LC_CALCULATION_TIMEOUT_ERROR)).toEqual('/common//img/statuses/female-error')
            expect(getPathImage(STATUS.LC_DECLINED)).toEqual('/common//img/statuses/female-error')
            expect(getPathImage(STATUS.LC_ERROR)).toEqual(`/lib.creditability/${version}/assets/images/ruler`)
            expect(getPathImage(STATUS.LC_EXPIRED)).toEqual('/common//img/statuses/female-info')
            expect(getPathImage(STATUS.LC_NEED_MORE_INFORMATION)).toEqual('/common//img/statuses/female-info')
            expect(getPathImage(STATUS.LC_NOT_CALCULATED)).toEqual(`/lib.creditability/${version}/assets/images/ruler`)
            expect(getPathImage('SOME_STRANGE_STATUS')).toEqual(`/lib.creditability/${version}/assets/images/ruler`)
        })

        it('should renders correctly with scale and button', () => {
            const onClickSpy = jest.fn()
            const wrapper = shallow(<CapacityProductDumb
                status="LC_ACCEPTED"
                system="Reactive"
                locales={{
                    title: 'Кредитный потенциал рассчитан',
                    description: 'Узнать какая сумма вам одобрена',
                    button: 'Узнать'
                }}
                parts={{
                    available: 2000,
                    used: 500,
                    reserved: 250
                }}
                link="/loans/creditability"
                onButtonClick={onClickSpy}
            />)

            const button = wrapper.find(Link)
            button.simulate('click')
            expect(wrapper).toMatchSnapshot()
            expect(onClickSpy).toHaveBeenCalled()
        })

        it('should renders correctly without scale and button', () => {
            const wrapper = shallow(<CapacityProductDumb
                status="LC_DECLINED"
                locales={{
                    title: 'Кредитный потенциал рассчитан',
                    description: 'Узнать результат расчёта'
                }}
                link="/loans/creditability"
                onButtonClick={void 0}
            />)

            expect(wrapper).toMatchSnapshot()
        })
    })
})
