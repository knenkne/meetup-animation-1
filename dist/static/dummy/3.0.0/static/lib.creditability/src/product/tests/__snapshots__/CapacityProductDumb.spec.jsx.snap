// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapacityProduct /> <CapacityProductDumb /> should renders correctly with scale and button 1`] = `
<Fragment>
  <div
    className="header"
  >
    <div
      className="scaleContainer"
    >
      <CapacityProductScale
        available={2000}
        reserved={250}
        used={500}
      />
    </div>
    <h3
      className="title"
    >
      Кредитный потенциал рассчитан
    </h3>
  </div>
  <img
    alt=""
    className="status femaleDone"
    role="presentation"
    src="/common//img/statuses/female-done.png"
    srcSet="/common//img/statuses/female-done@2x.png 2x, /common//img/statuses/female-done@3x.png 3x"
  />
  <div
    className="advice"
  >
    Узнать какая сумма вам одобрена
  </div>
  <Link
    as={[Function]}
    bold={false}
    colorScheme="button"
    external={false}
    href="/loans/creditability"
    mode=""
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    size="lg"
    theme={
      Object {
        "link": "button",
      }
    }
  >
    Узнать
  </Link>
</Fragment>
`;

exports[`<CapacityProduct /> <CapacityProductDumb /> should renders correctly without scale and button 1`] = `
<Fragment>
  <div
    className="header"
  >
    <div
      className="scaleContainer"
    />
    <h3
      className="title"
    >
      Кредитный потенциал рассчитан
    </h3>
  </div>
  <img
    alt=""
    className="status femaleError"
    role="presentation"
    src="/common//img/statuses/female-error.png"
    srcSet="/common//img/statuses/female-error@2x.png 2x, /common//img/statuses/female-error@3x.png 3x"
  />
  <div
    className="advice"
  >
    Узнать результат расчёта
  </div>
</Fragment>
`;
